designs/Transitive-Dependencies.md

```diff
@@ -0,0 +1,150 @@
+
+## Why: How do we know this is a real problem and worth solving?
+
+Knowing how a package is being included in a project helps a developer diagnose many problems. It also provides them with an opportunity to understand complex dependency chains they may have within their projects. 
+There is not a great solution that exists to understand the nature of top-level packages and their transitive dependencies.
+The solution explorer in Visual Studio does the best job at allowing a user to dive into each NuGet package & see all transitive dependencies from the top-level dependency.
+
+![](../resources/TransitiveDependencies/SolutionView.png)
+
+The dotnet CLI does not provide the insight into “why” a transitive dependency is listed, although it does list everything that has been resolved.
+
+![](../resources/TransitiveDependencies/DotNetCLI.png)
+
+The Visual Studio Package Manager UI does not support the concept of showing transitive packages in any fashion.
+
+![](../resources/TransitiveDependencies/VSPMUI.png)
+
+## Success: How do we know if we have solved this problem?
+
+By measuring the number of detractors regarding transitive dependencies & promoters of this new experience in future NuGet client surveys.
+
+We will add telemetry to categorize when a top-level package is installed, and whether it was done in the context of promoting a transitive package to a top-level one.
+
+## Audience: Who are we building this for?
+
+We are building this for .NET users, both old and new, to be successful with understanding the types of NuGet dependencies in their project(s) and solution(s).
+
+## What: What does this look like in the product?
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+## Context and Scope
+
+-	https://github.com/NuGet/Home/issues/5887 
+-	https://developercommunity.visualstudio.com/content/idea/538119/add-dependency-tree-view-for-packagereferences.html 
+
+## Minimal Requirements
+
+-	Visual list of transitive dependencies.
+-	Ability to see a transitive dependency package details.
+-	Ability to determine where a transitive dependency originated.
+-	Ability to promote a transitive dependency to a top-level dependency.
+
+## Scenarios and User Experience
+
+### CLI
+
+We should consider a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of where the package came from. This could be a new column next to “Resolved” which says “Transitively Referenced” or “Referenced” and has a list of the top-level packages that requested the dependency.
+
+![](../resources/TransitiveDependencies/TransitiveDotNetCLI.png)
+
+### Visual Studio
```

Comment:
Accessibility should be called out. If we have any input from the a11y team, let's add it as well. At a minimum, note here that we will run checks in Accessibility Insights, and ensure screen-readers can differentiate Top/Transitive, as well as read the tooltip content.

Reply:
Accessibility concerns should've been called out regarding the tooltip aspect. We will do a full compatibility pass when we're nearing the craftsmanship review of the UX board. It's likely we will have to cut the tooltip given we have no alternatives & need to unblock package vulnerability work for now.

Reply:
Was resolved but I don't see the details added to the spec. If there will be a technical spec coming after this, let me know.

Reply:
I didn't add any details to this spec as it's a design document. Although accessibility is considered & called out in this spec already, we don't have any insight or clarity into further accessibility concerns right now & those may need to be considered in the technical implementation as these are more test/implementation scenarios than design scenarios.

Reply:
/cc @drewgillies if you were planning to write-up anything for technical spec to make note of accessibility during implementation.